modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -289,16 +284,40 @@ const DataSet = ({
 
 	useEffect(() => {
 		setDataLoading(true);
+		setApiError(null);
 
-		requestData().then((data) => {
+		requestData().then(({data, ok, status}) => {
 			if (isMounted()) {
-				updateDataSetItems(data);
-
+				if (!ok) {
+					setApiError({data, status});
+				}
+				else {
+					updateDataSetItems(data);
+				}
 				setDataLoading(false);
 			}
 		});
 	}, [isMounted, requestData, setDataLoading]);
 
+	useEffect(() => {
+		if (isMounted() && apiError) {
+			const {data, status: statusCode} = apiError;
+
+			const apiErrorTitle = `${Liferay.Language.get(
+				'error'
+			)} ${statusCode}`;
+			const apiErrorMessage = `${data.status}, ${data.title}`;
+
+			logError(apiErrorMessage);
+
+			openToast({
+				message: apiErrorMessage,
+				title: apiErrorTitle,
```

Comment:
Hey @beltranrengifo, `apiErrorTitle` is being used in just one place. Could you inline it? 💔